components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/authenticator/uaepass/UAEPassAuthenticator.java

```diff
@@ -0,0 +1,958 @@
+     *
+     * @param authenticatorProperties  The user input fields of the authenticator.
+     * @param loginPage                Current authorize URL.
+     * @return authzUrl                Returns the modified authorized URL appending the additional query params.
+     */
+    public String processAdditionalQueryParamSeperation(Map<String, String> authenticatorProperties, String loginPage)
+        throws UAEPassAuthnFailedException {
+
+        String additionalQueryParams = authenticatorProperties.get(UAEPassAuthenticatorConstants.UAE.QUERY_PARAMS);
+        String[] splittedQueryParamsArr;
+
+        if (additionalQueryParams.contains(",")) {
+            splittedQueryParamsArr = additionalQueryParams.split(",");
+        } else {
+            splittedQueryParamsArr = additionalQueryParams.split("&");
+        }
+
+        String[] keyValuePairs;
+        Map<String, String> paramMap = new HashMap<>();
+
+        for (int i = 0; i < splittedQueryParamsArr.length; i++) {
+            keyValuePairs = (splittedQueryParamsArr[i]).split("=");
+            paramMap.put(keyValuePairs[0], keyValuePairs[1]);
+        }
+        String finalAuthzUrl = null;
+        try {
+            String authzUrl = FrameworkUtils.buildURLWithQueryParams(loginPage, paramMap);
+            if (!(authzUrl.contains(UAEPassAuthenticatorConstants.UAE.ACR_VALUES))) {
+                paramMap.put(UAEPassAuthenticatorConstants.UAE.ACR_VALUES,
+                        UAEPassAuthenticatorConstants.UAEPassRuntimeConstants.DEFAULT_ACR_VALUES);
+            }
+            if (!(authzUrl.contains(UAEPassAuthenticatorConstants.UAE.SCOPE))) {
+                paramMap.put(UAEPassAuthenticatorConstants.UAE.SCOPE,
+                        UAEPassAuthenticatorConstants.UAEPassRuntimeConstants.DEFAULT_SCOPES);
+            }
+            finalAuthzUrl = FrameworkUtils.buildURLWithQueryParams(loginPage, paramMap);
+        } catch (IllegalArgumentException | UnsupportedEncodingException e) {
+            if (LOG.isDebugEnabled()) {
+                LOG.error("Authorize URL creation failed with the additional query parameters issue.");
+            }
+            throw new UAEPassAuthnFailedException("Authentication process failed. Unable to set " +
+                    "additional query parameters to the authorize request.", e);
+        }
+
+        return finalAuthzUrl;
+    }
+
+    /**
+     * This method is used to retrieve user claims as key value pairs to the Java Map object from user info endpoint.
+     *
+     * @param oAuthResponse         The response from OAuthClient to authenticator by the UAEPass.
+     *                              (Use to teh get access token)
+     * @param context               The Authentication context received by authenticator.
+     * @return Map<String, Object>  Map object of key value pairs of the logged user.
+     */
+    public Map<String, Object> getUserInfoUserAttributes(OAuthClientResponse oAuthResponse,
```

Comment:
Do we need to keep this method as public

Reply:
Fixed with : [c9f22571bc5](https://github.com/wso2-extensions/identity-outbound-auth-uaepass/pull/1/commits/c9f22571bc5ffd21428c8b3a022c61e2045de40f)